File: criteria-api/src/test/java/example/criteria_api/BatchDeleteTest.java
package example.criteria_api;

import example.criteria_api.repository.EmployeeRepository;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

@ExtendWith(TestEnvironment.class)
public class BatchDeleteTest {

  private final EmployeeRepository repository;

  BatchDeleteTest(DbConfig config) {
    repository = new EmployeeRepository(config);
  }

  @Test
  public void testBatchDelete() {
    var list = repository.selectAll();
    repository.batchDelete(list);
  }
}

File: dsl-style-kotlin/src/test/kotlin/example/dsl/style/kotlin/BatchDeleteTest.kt
package example.dsl.style.kotlin

import example.dsl.style.kotlin.repository.EmployeeRepository
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.seasar.doma.jdbc.Config

@ExtendWith(TestEnvironment::class)
class BatchDeleteTest internal constructor(config: Config) {

    private val repository: EmployeeRepository = EmployeeRepository(config)

    @Test
    fun testBatchDelete() {
        val list = repository.selectAll()
        repository.batchDelete(list)
    }
}
